needle/core/utils/utils.py

```diff
@@ -40,6 +40,11 @@ def extract_paths_from_string(str):
         # Error
         return None, None
 
+    @staticmethod
+    def get_file_path(file_str):

```

Comment:
rename in something more relevant, like `extract_directory_from_path`. Or something like this